classes/WooCommerce.php

```diff
@@ -85,13 +85,12 @@ public static function show_badge_on_variation_select() {
 		<script type="text/javascript">
             (function ($) {
                 let $form         = $('form.variations_form');
-                let $product      = $form.closest('.product');
+                let $product      = $('div.woocommerce-product-gallery');
```

Comment:
Altough it would work in native WooCommerce, I'm not sure it would work in all themes/builders... (I'm pretty sure certain configs just have .product class). I'd prefer a multiple selector, or a check... Also we shoud keep the $form.closest for the same reason

Reply:
Understood, and you certainly have more experience on this than i do. This is what i had to do to make this work with one builder-like environment: kadence wootemplate feature (which is part of kadence offer, roughly a gutenberg based template environment to create custom single product pages; since the layout is arbitrary, the only common parent with the form is arbirary and does not posess the woocommerce/product class: this is why is targeted directly the gallery.